core/src/dbs/mod.rs

```diff
@@ -21,17 +21,37 @@ pub mod capabilities;
 pub mod node;
 
 pub use self::capabilities::Capabilities;
+pub(crate) use self::executor::*;
+pub(crate) use self::iterator::*;
 pub use self::notification::*;
 pub use self::options::*;
 pub use self::response::*;
 pub use self::session::*;
-
-pub(crate) use self::executor::*;
-pub(crate) use self::iterator::*;
 pub(crate) use self::statement::*;
 pub(crate) use self::variables::*;
+#[cfg(not(target_arch = "wasm32"))]
+use crate::err::Error;
+#[cfg(not(target_arch = "wasm32"))]
+use tokio::sync::oneshot::error::RecvError;
+#[cfg(not(target_arch = "wasm32"))]
+use tokio::task;
 
 #[doc(hidden)]
 pub mod fuzzy_eq;
 #[cfg(test)]
 pub(crate) mod test;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(super) async fn spawn_blocking<T, F, E>(f: F, e: E) -> Result<T, Error>
+where
+	F: FnOnce() -> Result<T, Error> + Send + 'static,
+	E: FnOnce(RecvError) -> Error,
+	T: Send + 'static,
+{
+	let (send, recv) = tokio::sync::oneshot::channel();
+	task::spawn_blocking(move || {
+		let res = f();
+		let _ = send.send(res);
+	});
+	recv.await.map_err(e)?
```

Comment:
Why do you use a channel here? task::spawn_blocking returns a join handle which you can await to get the returned value of a closure.

Reply:
I think I've been overthinking this, reading this:
https://ryhl.io/blog/async-what-is-blocking/

Indeed, `spawn_blocking` is sufficient.